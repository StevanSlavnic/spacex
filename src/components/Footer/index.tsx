export default function Footer() {
  return (
    <footer className="bg-gray-800 p-4">
      <div className="container mx-auto flex justify-center items-center">
        <p className="text-white text-lg">&copy; 2024 SpaceX Launches</p>
      </div>
    </footer>
  );
}
