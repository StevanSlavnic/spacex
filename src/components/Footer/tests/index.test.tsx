import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Footer from "../index";

describe("Footer component", () => {
  it("renders without crashing", () => {
    const { container, getByText } = render(<Footer />);
    expect(getByText(/© 2024 SpaceX Launches/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has correct class names", () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toHaveClass("bg-gray-800 p-4");
  });

  it("contains a paragraph with the correct text", () => {
    const { getByText } = render(<Footer />);
    const paragraph = getByText(/© 2024 SpaceX Launches/i);
    expect(paragraph).toBeInTheDocument();
    expect(paragraph).toHaveClass("text-white text-lg");
  });
});
