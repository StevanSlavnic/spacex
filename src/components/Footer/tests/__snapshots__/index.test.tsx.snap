// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders without crashing 1`] = `
<footer
  class="bg-gray-800 p-4"
>
  <div
    class="container mx-auto flex justify-center items-center"
  >
    <p
      class="text-white text-lg"
    >
      Â© 2024 SpaceX Launches
    </p>
  </div>
</footer>
`;
